packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapsPlugin.java

```diff
@@ -181,5 +194,88 @@ public void onActivityDestroyed(Activity activity) {
     public Lifecycle getLifecycle() {
       return lifecycle;
     }
+
+    @NonNull
+    @Override
+    public SavedStateRegistry getSavedStateRegistry() {
+      return savedStateRegistryController.getSavedStateRegistry();
+    }
+  }
+
+  /**
+   * This logic is incredibly hacky, but is necessary to address multiple problems with the Flutter
+   * embedding library.
+   *
+   * <p>Ideally, every {@link LifecycleOwner} that drives this plugin would also implement {@link
+   * SavedStateRegistryOwner}. This would make this class completely unnecessary. However, {@link
+   * io.flutter.embedding.android.FlutterActivity} does not (at the time of this writing) do this.
+   *
+   * <p>Given that, ideally, every LifecycleOwner that drives this plugin would restore state at the
+   * correct time. This would allow us to consistently drive state restoration via {@link
+   * OnSaveInstanceStateListener} rather than possibly needing to delegate to the LifecycleOwner.
+   * Unfortunately, {@link io.flutter.embedding.android.FlutterFragment} (at the time of this
+   * writing) restores state in {@link androidx.fragment.app.Fragment#onActivityCreated(Bundle)}
+   * rather than in {@link androidx.fragment.app.Fragment#onCreate(Bundle)}. Note that this is a
+   * pretty serious bug in FlutterFragment.
```

Comment:
@xster FYI

Reply:
I've got a separate PR pending to fix this and a few other lifecycle timing bugs: https://github.com/flutter/engine/pull/22203